### PR TITLE
Refactor Ta Assignment in Criterion Model

### PR DESCRIPTION
**Pull request by ashih2018**
Updated Criterion model (and small changes to other files) to pass many of the failing tests in `rubric_criterion_spec.rb` as well as the test on line 259 in `criterion.rb`:
https://github.com/ashih2018/Markus/blob/2814c48756cd84f2c76049dc2e99bdea13274a0d/spec/support/criterion.rb#L259-L279

## Motivation and Context
After @david-yz-liu performed the single table inheritance migration by removing the Rubric, Checkbox, and Flexible Criterion, this PR addresses references to the old 'three-table format' and refactors them using the new single table format. Specifically, I mostly refactored code pertaining to assigning TAs in `criterion.rb` as well as changed other files to fix test cases in `rubric_criterion_spec.rb`

## Your Changes
- Updated rubric criterion factory
- Removed most polymorphic references in the criterion models as well as `criterion_ta_associations`
- Updated `self.assign_tas` method as well as other methods in `criterion.rb` file.

**Description**:
Refactored code to reflect single table structure in database.

**Type of change** (select all that apply):
- [x] Refactoring (internal change to codebase, without changing functionality)
- [x] Test update (change that modifies or updates tests only)

## Testing
If you run `bundle exec rspec ./spec/models/rubric_criterion_spec.rb`, there should be expected 65 examples with 24 failures.

## Questions and Comments (if applicable)
I will continue to refactor the methods regarding to assigning tas as I continue to refactor the code to reflect the single table model in the backend. For example `self.assign_all_tas` in the Criterion model currently expects its argument in the form `[[id], ta_id]`, where as we can simplify it to `[id, ta_id]`

## Checklist
- [x] I have performed a self-review of my own code.
- [x] I have fixed any Hound bot comments (check after opening pull request).
- [x] I have verified that the TravisCI tests have passed (check after opening pull request).
- [x] I have reviewed the test coverage changes reported on Coveralls (check after opening pull request).

### CODE REVIEW

File: app/models/criterion.rb [hunk 1]
```diff
@@ -25,26 +34,33 @@ class Criterion < ApplicationRecord
...
   # must belong to the given assignment +assignment+.
-  def self.randomly_assign_tas(criterion_ids_types, ta_ids, assignment)
-    assign_tas(criterion_ids_types, ta_ids, assignment) do |criterion_ids, criterion_types, ta_ids|
+  def self.randomly_assign_tas(criterion_ids, ta_ids, assignment)
+    assign_tas(criterion_ids, ta_ids, assignment) do |c_ids, t_ids|
       # Assign TAs in a round-robin fashion to a list of random criteria.
-      crit_ids_and_types = criterion_ids.zip(criterion_types).shuffle
-      crit_ids_and_types.zip(ta_ids.cycle).map &:flatten
+      shuffled_criterion_ids = c_ids.shuffle
+      shuffled_criterion_ids.zip(t_ids.cycle).map &:flatten
```
[the remark this answers is not in the document]
Comment (ashih2018) [reply app/models/criterion.rb:52]: what should I be parenthesizing here?

[the remark this answers is not in the document]
Comment (mishaschwartz) [reply app/models/criterion.rb:52]: `shuffled_criterion_ids.zip(t_ids.cycle).map(&:flatten)`

[the remark this answers is not in the document]
Comment (ashih2018) [reply app/models/criterion.rb:52]: @mishaschwartz  thanks!

File: app/models/criterion.rb [hunk 2]
```diff
@@ -89,23 +101,15 @@ def self.assign_tas(criterion_ids_types, ta_ids, assignment)
     # CriterionTaAssociation.create when the PG driver supports bulk create,
     # then remove the activerecord-import gem.
     CriterionTaAssociation.import(columns, values, validate: false)
-
     Grouping.update_criteria_coverage_counts(assignment)
-    criterion_ids_by_type = {}
```
Comment (ashih2018) [inline app/models/criterion.rb:94]: `criterion_ids_by_type` was not referenced anywhere so I removed it

**Pull request merged**